examples/http-compression/http_compression.bal

```diff
@@ -0,0 +1,92 @@
+import ballerina/http;
+import ballerina/log;
+
+endpoint http:Listener listenerEndpoint {
+    port: 9090
+};
+
+// Since compression behaviour of service is set as COMPRESSION_AUTO, entity body compression is done according
+// to the scheme indicated in Accept-Encoding request header. When particular header is not present or the
+// header value is "identity", compression is not performed.
+@http:ServiceConfig {
+    compression: {
+        enable: http:COMPRESSION_AUTO
+    }
+}
+service<http:Service> autoCompress bind listenerEndpoint {
+
+    @http:ResourceConfig {
+        path: "/"
+    }
+    invokeEndpoint(endpoint caller, http:Request req) {
+        http:Response response = new;
+        response.setJsonPayload({ "Type": "Auto compression" });
+        caller->respond(response) but { error e => log:printError("Error sending response", err = e) };
```

Comment:
Since we are not passing any headers shall we directly set the payload to the body?
caller -> respond({ "Type": "Auto compression" });